src/libpgmoneta/json.c

```diff
@@ -620,6 +620,24 @@ parse_string(char* str, uint64_t* index, struct json** obj)
          // The key
          while (idx < len && str[idx] != '"')
```

Comment:
We should have more security check about this now in case there's a `"` in the middle that's missing preceding `\`

Reply:
I think the rest of the code logic will take care of it unless someone deliberately introduce some weird cases like-

**(Extreme cases)**

`{"na":"me"}":"ashutosh"}` 
where expected key - `na":"me"}` and value is `ashutosh` 
while we will get key - `na` and value - `me`

or

`{"na":"me"," : ":"ashutosh"}`
where he expects key - `na":"me` and value - `:"ashutosh`
but will get key1 - `na`, value1 - `me` and key2 - `:`, value2 - `ashutosh`

While the second case is not invalid. The above case can be mitigated by parenthesis matching which is O(n).

@Jubilee101 Any thoughts on this

Reply:
> {"na":"me"}":"ashutosh"}

Current logic is more of a "max effort" way, that it parses as much as it can, and stops when it has finshed parsing an entire json item (when it sees the enclosing `}`), regardless of if there are more gibberish left to parse. The fix should be easy, if [the parsing has finished](https://github.com/pgmoneta/pgmoneta/blob/b1f97d07a8ea71a9da0d5517d9716d104d0e1d1f/src/libpgmoneta/json.c#L539) but there are still `!isspace()` characters left, the function should error out. You can try and see if it solves the problem, otherwise we open a separate issue to address it.

> {"na":"me"," : ":"ashutosh"}

Nah this is a valid case, if you forget to use `\` and end up getting something unexpected, we are not responsible for that. 

I was mostly thinking about simpler cases like `"ab"c":"DE"`, if the later logic takes care of it then we can just leave it.

Reply:
Hmm, will try that